### PR TITLE
*: Make column test stable and fix prepare statement issue

### PR DESCRIPTION
**Pull request by zimulala**
ddl: make column test stable
*: fix prepare statement issue when retrying

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

File: session.go [hunk 1]
```diff
@@ -561,7 +565,7 @@ func (s *session) DropPreparedStmt(stmtID uint32) error {
 	if _, ok := vars.PreparedStmts[stmtID]; !ok {
 		return executor.ErrStmtNotFound
 	}
-	delete(vars.PreparedStmts, stmtID)
+	vars.RetryInfo.DroppedPreparedStmtIDs = append(vars.RetryInfo.DroppedPreparedStmtIDs, stmtID)
```
Comment (coocood) [inline session.go]: In retry, `ExecutePreparedStmt` will execute with the same stmtID, so we should not dropping the preparedStmts until finished the transaction, dropping it in ClearRetryInfo.

Comment (shenli) [inline session.go]: Will we retry prepare statement? Or we only retry execute statement?

Comment (coocood) [inline session.go]: We only retry execute statement.
`Session.PrepareStmt` do not add statement history.

**Comment by shenli** [comment]
LGTM

**Pull request merged**